addons/logger_debug/tools/log_filter_debug.py

```diff
@@ -0,0 +1,42 @@
+
+import logging
+from threading import current_thread
+from typing import Dict, Set
+
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+class LogFilterDebugDB(logging.Filter):
+    def __init__(self):
+        super().__init__('LogFilterDebugDB')
+        self.logger_db: Dict[str, Set[str]] = {}
+        """A mapping of debug loggers to the databases they are enabled for.
+        e.g. {'odoo.addons.logger_debug': {'db1', 'db2'}}"""
+
+    def filter(self, record):
+        if record.levelno != logging.DEBUG or (ct_db := getattr(current_thread(), 'dbname', None)) is None:
+            return True
+        # TODO: Check how to handle the cases like IoT logs logger which have a debug level
+        return (logger_dbs := self.logger_db.get(record.name)) and ct_db in logger_dbs
```

Comment:
should we use .startswith() due to logger inheritance ? For example we set in debug logger A does A.B should log in debug as well ?